src/datasources/errors/http-error-factory.spec.ts

```diff
@@ -0,0 +1,58 @@
+import { AxiosError, AxiosResponse } from 'axios';
+import { HttpErrorFactory } from './http-error-factory';
+import { HttpExceptionPayload } from './interfaces/http-exception-payload.interface';
+
+describe('HttpErrorFactory', () => {
```

Comment:
nitpick: One additional test that can be written is if you get an error response with an `undefined` or `null` status.

In those cases we map it to a `500`